scripts/actions/print-matrix.clj

```diff
@@ -1,41 +1,56 @@
 #!/usr/bin/env bb
 
-(def default-clojure-version "") ;; set in project.clj
+;; determines the build matrix for the GitHub Actions build. 
+;; try it locally:
+;;   # normal builds
+;;   $ GITHUB_EVENT_NAME=pull_request ./scripts/actions/print-matrix.clj
+;;   $ GITHUB_EVENT_NAME=push ./scripts/actions/print-matrix.clj
+;;   # cron build
+;;   $ GITHUB_EVENT_NAME=schedule ./scripts/actions/print-matrix.clj
+
 (def default-java-version "11.0.9")
 (def java-15-version "15")
-(def clojure-next "1.10.2-rc1")
 (def non-cron-ctia-nsplits
   "Job parallelism for non cron tests."
   10)
 (def cron-ctia-nsplits
   "Job parallelism for cron tests."
   2)
 
+(defn splits-for [base nsplits]
+  {:pre [(pos? nsplits)]
+   :post [(= (range nsplits)
+             (map :this_split %))
+          (= #{nsplits}
+             (into #{} (map :total_splits) %))]}
+  (for [this-split (range nsplits)]
+    (assoc base
+           :this_split this-split
+           :total_splits nsplits)))
+
 (defn non-cron-matrix
   "Actions matrix for non cron builds"
   []
-  (for [this-split (range non-cron-ctia-nsplits)]
-    {:this_split this-split
-     :total_splits non-cron-ctia-nsplits
-     :clojure_version default-clojure-version
-     :java_version default-java-version}))
+  (splits-for
+    {:ci_profiles "default"
+     :java_version default-java-version}
+    non-cron-ctia-nsplits))
 
 (defn cron-matrix
   "Actions matrix for cron builds"
   []
-  (for [[clojure-version java-version] [[default-clojure-version default-java-version]
-                                        [clojure-next default-java-version]
-                                        [clojure-next java-15-version]]
-        this-split (range cron-ctia-nsplits)]
-    {:this_split this-split
-     :total_splits cron-ctia-nsplits
-     :clojure_version clojure-version
-     :java_version java-version}))
+  (mapcat #(splits-for % cron-ctia-nsplits)
+          (concat
+            [{:ci_profiles "default"
+              :java_version default-java-version}]
+            (map #(into {:ci_profiles "next-clojure"} %)
+                 [{:java_version default-java-version}
+                  {:java_version java-15-version}]))))
 
 (defn edn-matrix []
   {:post [(seq %)]}
   (if (= "schedule" (System/getenv "GITHUB_EVENT_NAME"))
     (cron-matrix)
     (non-cron-matrix)))
 
-(println "::set-output name=matrix::" (json/generate-string (edn-matrix) {:pretty false}))
+(println (str "::set-output name=matrix::" (json/generate-string (edn-matrix) {:pretty false})))
```

Comment:
looks like there used to be a single whitespace between `name=matrix::` and the json string; I assume the removal is intentional?

Reply:
Yes, the syntax is `::set-output name={name}::{value}`.